libs/core/src/lib/select/select.component.ts

```diff
@@ -58,15 +72,22 @@ export interface OptionStatusChange {
         '[class.fd-select-custom-class--mobile]': 'mobile',
     },
     providers: [
-        SelectProxy,
         {
-            provide: NG_VALUE_ACCESSOR,
-            useExisting: forwardRef(() => SelectComponent),
-            multi: true
-        }
+            provide: SELECT_COMPONENT,
```

Comment:
What is this? you cant really use useExisting here. You are creating a provider which will not work with Forms. there is a reason why its called NG_VALUE_ACCESOR. it is used internally by form API

Reply:
We probbale need to keep both providers: `NG_VALUE_ACCESOR` as well as `SELECT_COMPONENT`

Reply:
As we added selection model, and keyManagers as services adding NG_VALUE_ACCESOR and usage of ngControl  to display preselected values are resulting to cycled dependency , ng Control inject errors. To avoid that, I have provided SELECT_COMPONENT.

Reply:
So we need to fix somehow this `cycled dependency`.
We can't drop `NG_VALUE_ACCESOR` provider cause it wont work with angular forms

Reply:
Please suggest on this, 
I tried couple of ways like creating instance of ngControl, had both select_component and ng_value_accessor, tried only having ng_value_accessor etc and ended up with infinite error loops.

Reply:
Oh, I see.
So we don't need this provider since we have the linking in the constructor as :
```
constructor(@Optional() @Self() public ngControl: NgControl) {
        if (this.ngControl) {
            this.ngControl.valueAccessor = this;
        }
    }
```

Reply:
I missed this one, as we started to the same thing we have in platform code to connect it manually to forms api . thanks.